### PR TITLE
Improve TIAGo for better compatibility with real robot

### LINKED ISSUE #700: Improve ROS2 compatibility with real TIAGo robot
Currently the Webots ROS2 topics are difficult to use with a real TIAGo robot and it would be interesting to publish additional information.

_Roadmap: the plan would be to first start focusing on the camera-related topics and then to joints/navigation ones._

Example here on the implementation that was done for gazebo and all the program that is launched the launch file:
https://github.com/pal-robotics/tiago_simulation/blob/humble-devel/tiago_gazebo/launch/tiago_gazebo.launch.py

The real TIAGo is launching this file at startup:
https://github.com/pal-robotics/tiago_robot/blob/humble-devel/tiago_bringup/launch/tiago_bringup.launch.py

Here is the list of the ROS2 topics used by the TIAGo robot:

- Camera topics:
    - /head_front_camera/depth_registered/camera_info [sensor_msgs/msg/CameraInfo]
    - /head_front_camera/depth_registered/image_raw [sensor_msgs/msg/Image]
    - /head_front_camera/depth_registered/points [sensor_msgs/msg/PointCloud2]
    - /head_front_camera/rgb/camera_info [sensor_msgs/msg/CameraInfo]
    - /head_front_camera/rgb/image_raw [sensor_msgs/msg/Image]

 Camera_info:
```
header:
      stamp:
        sec: 35
        nanosec: 723000000
      frame_id: head_front_camera_rgb_optical_frame
    height: 480
    width: 640
    distortion_model: plumb_bob
    d:
    - 1.0e-08
    - 1.0e-08
    - 1.0e-08
    - 1.0e-08
    - 1.0e-08
    k:
    - 522.1910329546544
    - 0.0
    - 320.5
    - 0.0
    - 522.1910329546544
    - 240.5
    - 0.0
    - 0.0
    - 1.0
    r:
    - 1.0
    - 0.0
    - 0.0
    - 0.0
    - 1.0
    - 0.0
    - 0.0
    - 0.0
    - 1.0
    p:
    - 522.1910329546544
    - 0.0
    - 320.5
    - -0.0
    - 0.0
    - 522.1910329546544
    - 240.5
    - 0.0
    - 0.0
    - 0.0
    - 1.0
    - 0.0
    binning_x: 0
    binning_y: 0
    roi:
      x_offset: 0
      y_offset: 0
      height: 0
      width: 0
      do_rectify: false
```

- Essential topics
    - /arm_controller/joint_trajectory [trajectory_msgs/msg/JointTrajectory]
    - /arm_controller/state [control_msgs/msg/JointTrajectoryControllerState]
    - /arm_controller/transition_event [lifecycle_msgs/msg/TransitionEvent]
    - /clock [rosgraph_msgs/msg/Clock]
    - /gripper_controller/joint_trajectory [trajectory_msgs/msg/JointTrajectory]
    - /gripper_controller/state [control_msgs/msg/JointTrajectoryControllerState]
    - /gripper_controller/transition_event [lifecycle_msgs/msg/TransitionEvent]
    - /head_controller/joint_trajectory [trajectory_msgs/msg/JointTrajectory]
    - /head_controller/state [control_msgs/msg/JointTrajectoryControllerState]
    - /head_controller/transition_event [lifecycle_msgs/msg/TransitionEvent]
    - /joint_states [sensor_msgs/msg/JointState]
    - /joy [sensor_msgs/msg/Joy]
    - /joy_priority [std_msgs/msg/Bool]
    - /joy_vel [geometry_msgs/msg/Twist]
    - /key_vel [geometry_msgs/msg/Twist]
    - /mobile_base_controller/cmd_vel_unstamped [geometry_msgs/msg/Twist]
    - /mobile_base_controller/odom [nav_msgs/msg/Odometry]
    - /mobile_base_controller/transition_event [lifecycle_msgs/msg/TransitionEvent]
    - /nav_vel [geometry_msgs/msg/Twist]
    - /parameter_events [rcl_interfaces/msg/ParameterEvent]
    - /play_motion2/transition_event [lifecycle_msgs/msg/TransitionEvent]
    - /robot_description [std_msgs/msg/String]
    - /rosout [rcl_interfaces/msg/Log]
    - /rviz_joy_vel [geometry_msgs/msg/Twist]
    - /scan_raw [sensor_msgs/msg/LaserScan]
    - /tf [tf2_msgs/msg/TFMessage]
    - /tf_static [tf2_msgs/msg/TFMessage]
    - /torso_controller/joint_trajectory [trajectory_msgs/msg/JointTrajectory]
    - /torso_controller/state [control_msgs/msg/JointTrajectoryControllerState]
    - /torso_controller/transition_event [lifecycle_msgs/msg/TransitionEvent]

### PR DESCRIPTION
**Pull request by ygoumaz**
Resolves #700.

Tasks:
- [x] Add RGB-D Camera (Astra) topics
- [x] Configure those topics to match real TIAGo names (keep backwards compatibility in `webots_ros2_driver`)
- [x] Replace `TIAGo_Iron` by `TIAGo_Lite` in navigation world file.
- [x] Update original launch file `robot_launch.py` to start `ros2_control` controllers **after** Webots is ready (This change should be applied to all packages, see #719).
- [x] Duplicate world with normal `TIAGo` robot (`default_bringup.wbt`).
- [x] New launch file `robot_bringup_launch.py` to start TIAGo (in the new world) with all the packages compatible with real robot (`tiago_bringup` package).
- [x] New simplified `ros2_control_bringup` parameters file (parameters for each controller are directly taken from the `tiago_bringup` package).
- [x] New URDF file `tiago_bringup_webots.urdf` to interface all joints with started controllers (head, torso, base, gripper, arm).
- [x] New simplified rviz configuration file (`default_bringup.rviz`) to remove all navigation parts.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
All the camera related topics work fine.
But when starting the example all the rest of the topics listed in #700 are still not available, is this correct?

Is this command (from the examples page) still up-to-date?
```ros2 launch webots_ros2_tiago robot_launch.py rviz:=true slam:=true```

In RVIZ I see only the base but not the body nor the map.
Screenshot from 2023-05-01 10-36-52

**Comment by ygoumaz** [comment]
This command is not up-to-date anymore. I will update it once the PR is merged. To start the full configuration of the TIAGo with all related topics, you should use the following command:

```bash
ros2 launch webots_ros2_tiago robot_bringup_launch.py rviz:=true
```

Note that you can also ignore RViz if you don't want to start it. You also need to install `tiago-bringup` to start all topics:
```bash
sudo apt install ros-humble-tiago-bringup
```

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

Now I see that all the topics are available and in RVIZ I can see the full robot in RVIZ.
I still don't have the map and I can no longer move teleoperating the robot as described in the old example but I will wait for the updated example description.

**Comment by ygoumaz** [comment]
As explained this morning, I have separated both examples. The `robot_launch.py` file is used to start the original example (featuring TIAGo Lite) with SLAM and Navigation. On the other hand, `robot_bringup_launch.py` is used to start the TIAGo with all the topics. Starting everything together is quite a big mess. I think it is clearer to the user if we provide two clear and simple examples and they can always adapt the launch files to add more packages if they need them.

**Comment by stefaniapedrazzi** [comment]
I will check again once the wiki has been update to see if I missed something, but as I reported previously when starting the old/original launcher, the robot visualization and the SLAM map were not displayed correctly.

**Comment by stefaniapedrazzi** [comment]
Thank you!
I confirm that all the three examples (navigation, Slam Toolbox, and cartographer) works correctly also for me.

File: webots_ros2_tiago/launch/robot_bringup_launch.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    world = LaunchConfiguration('world')
+    mode = LaunchConfiguration('mode')
+
+    webots = WebotsLauncher(
+        world=PathJoinSubstitution([package_dir, 'worlds', world]),
+        mode=mode,
+        ros2_supervisor=True
+    )
+
+    # The following line is important!
```
Comment (omichel) [inline webots_ros2_tiago/launch/robot_bringup_launch.py:103]:
```suggestion
    # The following lines are important!
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you.

**Review by stefaniapedrazzi — APPROVED** [review]

**Comment by ygoumaz** [comment]
Tests will run indefinitely until https://github.com/cyberbotics/webots/pull/6113 is merged.

**Pull request merged**